mbus/mbus-serial.c

```diff
@@ -246,7 +246,10 @@ mbus_serial_data_free(mbus_handle *handle)
             return;
         }
 
-        free(serial_data->device);
+        if (serial_data->device == NULL)
+        {
+            free(serial_data->device);
```

Comment:
Freeing a NULL pointer is generally a bad idea. I assume you want to do the opposite.